app/Http/Controllers/Api/Auth/RegisterController.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace App\Http\Controllers\Api\Auth;
+
+use App\Http\Controllers\Controller;
+use Illuminate\Http\Exceptions\HttpResponseException;
+use Illuminate\Http\Request;
+use App\User;
+use Illuminate\Support\Facades\App;
+use Illuminate\Support\Facades\Validator;
+
+class RegisterController extends Controller
+{
+    /**
+     * Register api
+     *
+     * @return \Illuminate\Http\Response
+     */
+
+    public function register(Request $request)
+    {
+
+        $validator = Validator::make(
+            $request->all(),
+            [
+                'name' => 'required',
+                'email' => 'required|email',
+                'password' => 'required',
```

Comment:
No pattern ? min/max length, lower/upper ? number etc. ?

Reply:
Laravel match automatically with Model and no need pattern for email as the validator takes charge of it.

Reply:
![gbkh3cj - imgur](https://user-images.githubusercontent.com/3429798/32701957-53741bb4-c7df-11e7-8963-617c0789e44f.gif)

Dumb question why email has 'required|**email** ?

Reply:
To validate against an email pattern actually 😛